### PR TITLE
src: mavlink: Try to reconnect if lost connection.

### LINKED ISSUE #108: mavlink-camera-manger ended up with corrupted buffers when TCP disconnects
Using TCP MAVLink endpoint, it is getting corrupted IO buffer errors on TCP server disconnection, never recovering itself.

Error:

```bash
ERROR 15:38:32.595 src/mavlink/mavlink_camera.rs:784: Error receiving a message as 1:102. Reason: Io(                                                                                               
    Error {                                                                                                                                                                                         
        kind: UnexpectedEof,                                                                                                                                                                        
        message: "failed to fill whole buffer",                                                                                                                                                     
    },                                                                                                                                                                                              
). Camera: MavlinkCameraInformation {                                                                                                                                                               
    component: MavlinkCameraComponent {                                                                                                                                                             
        system_id: 1,                                                                                                                                                                               
        component_id: 102,                                                                                                                                                                          
        stream_id: 1,                                                                                                                                                                               
        vendor_name: "H264 USB Camera: USB Camera",                                                                                                                                                 
        model_name: "batata1",                                                                                                                                                                      
        firmware_version: 0,                                                                                                                                                                        
        resolution_h: 1080,                                                                                                                                                                         
        resolution_v: 1920,                                                                                                                                                                         
        framerate: 30.0,                                                                                                                                                                            
        bitrate: 5000,                                                                                                                                                                              
        rotation: 0,                                                                                                                                                                                
        hfov: 90,                                                                                                                                                                                   
        thermal: false,                                                                                                                                                                             
    },                                                                                                                                                                                              
    mavlink_connection_string: "tcpout:127.0.0.1:5777",                                                                                                                                             
    mavlink_stream_type: VIDEO_STREAM_TYPE_RTPUDP,                                                                                                                                                  
    video_stream_uri: Url {                                                                                                                                                                         
        scheme: "udp",                                                                                                                                                                              
        cannot_be_a_base: false,                                                                                                                                                                    
        username: "",                                                                                                                                                                               
        password: None,                                                                                                                                                                             
        host: Some(                                                                                                                                                                                 
            Domain(                                                                                                                                                                                 
                "192.168.2.1",                                                                                                                                                                      
            ),                                                                                                                                                                                      
        ),                                                                                                                                                                                          
        port: Some(                                                                                                                                                                                 
            5601,                                                                                                                                                                                   
        ),                                                                                                                                                                                          
        path: "",
        query: None,
        fragment: None,
    },
    video_source_type: Local(
        VideoSourceLocal {
            name: "H264 USB Camera: USB Camera",
            device_path: "/dev/video7",
            typ: Usb(
                "usb-3f980000.usb-1.5",
            ),
        },
    ),
}
```

### PR DESCRIPTION
**Pull request by joaoantoniocardoso**
Fixes #108

### CODE REVIEW

**Comment by joaoantoniocardoso** [comment]
I've just saw a problem in the logic. I'll fix it asap.

**Comment by joaoantoniocardoso** [comment]
@patrickelectric its finally working :)

File: src/mavlink/mavlink_camera.rs [hunk 1]
```diff
@@ -141,8 +144,9 @@ impl Drop for MavlinkCameraComponent {
         // Remove id from used ids
         let id = self.component_id - mavlink::common::MavComponent::MAV_COMP_ID_CAMERA as u8;
         let mut vector = ID_CONTROL.lock().unwrap();
-        let position = vector.iter().position(|&vec_id| vec_id == id).unwrap();
-        vector.remove(position);
+        if let Some(position) = vector.iter().position(|&vec_id| vec_id == id) {
+            vector.remove(position);
+        }
```
Comment (patrickelectric) [inline src/mavlink/mavlink_camera.rs:149]: We should log when the id was not found

**Review by patrickelectric — APPROVED** [review]
LGTM
image

File: src/mavlink/mavlink_camera.rs (hunk 1 above)
Comment (joaoantoniocardoso) [reply src/mavlink/mavlink_camera.rs:149]: https://github.com/bluerobotics/mavlink-camera-manager/issues/111

**Pull request merged**